test/helpers/fakeSystem.js

```diff
@@ -69,8 +69,9 @@ const System = {
 			m.executed = true;
 			for (let i = 0; i < m.deps.length; i++) {
 				const dep = m.deps[i];
+				const setters = m.mod.setters[i];
 				System.ensureExecuted(dep);
-				m.mod.setters[i](System.registry[dep].exports);
+				if (setters) setters(System.registry[dep].exports);
```

Comment:
@joeldenning I just want to verify that I've read the documentation correctly: One could pass `undefined` as `setter` when not interested in the exports?

Reply:
This doesn't seem quite right to me, individual setters are supposed to be mandatory.

Rather though, it can be fine to omit the entire `setters` field when there are no imported bindings.

Reply:
I think you're both saying the same thing - omitting the setters property entirely is okay, but if it's provided you must provide a function for each setter.

The code on line 74 looks great to me.

```js
// Works
System.register(['external1'], function (_export, _context) {
  return {
    // omitting setters property entirely is completely legal
    execute() {
    }
  }
})
```

```js
// Doesn't work
System.register(['external1'], function (_export, _context) {
  return {
    setters: [
      // This is missing a setter function for external1, which is illegal.
    ],
    execute() {
    }
  }
})
```

Reply:
```js
// Legal
System.register(['external1'], function (_exports, _context) {
  return {
    // omitting entire setters property is fine
    execute() {}
  }
})
```

```js
// Legal
System.register(['external1'], function (_exports, _context) {
  return {
    // if setters property is provided, all setter functions must be provided
    setters: [
      function (ns) {
      }
    ]
    execute() {}
  }
})
```

```js
// Illegal
System.register(['external1'], function (_exports, _context) {
  return {
    setters: [
      // undefined, null, and any other non-functions are illegal
      undefined
    ]
    execute() {}
  }
})
```

Reply:
Right, exactly, but `setters` here is the individual item not the whole array.

Reply:
ok. Then I'll need to change that.